var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -56,37 +61,19 @@ class Scotch(Package):
     # from the Scotch hosting site.  These alternative archives include a
     # superset of the behavior in their default counterparts, so we choose to
     # always grab these versions for older Scotch versions for simplicity.
-    @when('@:6.0.0')
-    def url_for_version(self, version):
-        return '%s/scotch_%s_esmumps.tar.gz' % (Scotch.base_url, version)
-
-    @when('@6.0.1:')
     def url_for_version(self, version):
         return super(Scotch, self).url_for_version(version)
 
-    # NOTE: Several of the 'esmumps' enabled Scotch releases up to version
-    # 6.0.0 have broken build scripts that don't properly build 'esmumps' as a
-    # separate target, so we need a patch procedure to remove 'esmumps' from
-    # existing targets and to add it as a standalone target.
-    @when('@:6.0.0')
-    def patch(self):
-        makefile_path = os.path.join('src', 'Makefile')
-        with open(makefile_path, 'r') as makefile:
-            esmumps_enabled = any(re.search(r'^esmumps(\s*):(.*)$', line)
-                                  for line in makefile.readlines())
-
-        if not esmumps_enabled:
-            mff = FileFilter(makefile_path)
-            mff.filter(r'^.*((esmumps)|(ptesmumps)).*(install).*$', '')
-
-            mfesmumps_dir = os.path.dirname(os.path.realpath(__file__))
-            mfesmumps_path = os.path.join(mfesmumps_dir, 'Makefile.esmumps')
-            with open(makefile_path, 'a') as makefile:
-                makefile.write('\ninclude %s\n' % mfesmumps_path)
+    @when('+esmumps')
+    def url_for_version(self, version):
+        if version <= Version('6.0.0'):
+            self.versions[version]['md5'] = self.md5_esmumps[version]
+            return '%s/scotch_%s_esmumps.tar.gz' % (Scotch.base_url, version)
+        else:
+            return super(Scotch, self).url_for_version(version)
 
-    @when('@6.0.1:')
     def patch(self):
-        pass
+        self.configure()
```

Comment:
I like this change; I think that moving `configure` to the `patch` function makes a lot more sense.